doc/usage/extensions/autodoc.rst

```diff
@@ -548,6 +548,19 @@ There are also config values that you can set:
 
       New option ``'description'`` is added.
 
+.. confval:: autodoc_typehint_undoc
```

Comment:
Sorry for the late response. I missed your updates.

Now I'm thinking about the name of this option. I feel this option does not describe its behavior well. If my understanding is correct, it controls the target of the typehints generation to either all of parameters or documented parameters only when `autodoc_typehints = "description"`.

So I propose you the alternatives:

* `autodoc_typehints_description`
* `autodoc_typehints_description_target`
* `autodoc_typehints_description_params`

And it takes `"all"` or `"documented"` (`"documented_only"`?) to change the behavior. What do you think?

Reply:
Sure, I'm fine with that. I've updated the PR to call it `autodoc_typehints_description_target` and have it take either `"all"` or `"documented"`.